ibc-testkit/src/testapp/ibc/applications/nft_transfer/context.rs

```diff
@@ -0,0 +1,196 @@
+use ibc::apps::nft_transfer::context::{
+    NftClassContext, NftContext, NftTransferExecutionContext, NftTransferValidationContext,
+};
+use ibc::apps::nft_transfer::types::error::NftTransferError;
+use ibc::apps::nft_transfer::types::{
+    ClassData, ClassId, ClassUri, Memo, PrefixedClassId, TokenData, TokenId, TokenUri,
+};
+use ibc::core::host::types::identifiers::{ChannelId, PortId};
+use ibc::core::primitives::prelude::*;
+use ibc::core::primitives::Signer;
+
+use super::types::{DummyNft, DummyNftClass, DummyNftTransferModule};
+
+impl NftContext for DummyNft {
+    fn get_class_id(&self) -> ClassId {
+        "class_0".parse().expect("infallible")
+    }
+
+    fn get_id(&self) -> TokenId {
+        "token_0".parse().expect("infallible")
+    }
+
+    fn get_uri(&self) -> TokenUri {
+        "http://example.com".parse().expect("infallible")
+    }
+
+    fn get_data(&self) -> TokenData {
+        let data = r#"{"name":{"value":"Crypto Creatures"},"image":{"value":"binary","mime":"image/png"}}"#;
+        data.parse().expect("infallible")
+    }
+}
+
+impl NftClassContext for DummyNftClass {
+    fn get_id(&self) -> ClassId {
+        "class_0".parse().expect("infallible")
+    }
+
+    fn get_uri(&self) -> ClassUri {
+        "http://example.com".parse().expect("infallible")
+    }
+
+    fn get_data(&self) -> ClassData {
+        let data = r#"{"name":{"value":"Crypto Creatures"},"image":{"value":"binary","mime":"image/png"}}"#;
+        data.parse().expect("infallible")
+    }
+}
```

Comment:
Can we abstract `DummyNft` and `DummyNftClass` structs that encompasses the essential fields, so can call their fields for implementing these methods instead of hardcoding values?